### PR TITLE
fix(ui): preserve CJK scroll alignment

### PR DESCRIPTION
**Pull request by benvinegar**
## Summary

- Preserve the blank half-cell when horizontal scrolling starts inside a fullwidth CJK/emoji grapheme.
- Add regression coverage for `sliceTextByWidth("a日本b", 2, 4)` so subsequent content no longer shifts left.
- Document the user-visible fix in the changelog.

## Testing

- `bun test src/ui/lib/ui-lib.test.ts`
- `bun run typecheck`
- `bun run format:check`

This ports the still-relevant alignment edge case from #287 without taking the stale broader CJK implementation.

*This PR description was generated by Pi using OpenAI GPT-5*

### CODE REVIEW

File: src/ui/lib/ui-lib.test.ts [hunk 1]
```diff
@@ -314,7 +314,7 @@ describe("ui helpers", () => {
   test("text helpers measure and slice wide characters by terminal cells", () => {
     expect(measureTextWidth("日本語")).toBe(6);
     expect(sliceTextByWidth("a日本b", 1, 4)).toEqual({ text: "日本", width: 4 });
-    expect(sliceTextByWidth("a日本b", 2, 4)).toEqual({ text: "本b", width: 3 });
+    expect(sliceTextByWidth("a日本b", 2, 4)).toEqual({ text: " 本b", width: 4 });
     expect(fitText("日本語", 5)).toBe("日本.");
     expect(measureTextWidth(padText("日本", 6))).toBe(6);
   });
```
[the remark this answers is not in the document]
Comment (benvinegar) [reply src/ui/lib/ui-lib.test.ts:322]: Added the adjacent-wide-character case (`sliceTextByWidth("日本b", 3, 3)`) and a viewport-cap case (`sliceTextByWidth("日", 1, 1)`). I used a 2-cell wide glyph for the cap path because our terminal width model treats emoji grapheme clusters as width 2, not 4.

<sub><i>Responded by Pi using OpenAI GPT-5.</i></sub>

*This comment was generated by Pi using OpenAI GPT-5*
